File: rick.ts
import { exec } from "child_process";

const command = "curl ascii.live/rick";

exec(command, (error, stdout, stderr) => {
    if (error) {
        console.error(`Command failed: ${stderr}`);
        openRickroll();
        return;
    }
    
    console.log(stdout);  // Print command output
});

function openRickroll() {
    const url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";
    
    const platform = process.platform;
    if (platform === "win32") {
        exec(`start ${url}`);
    } else if (platform === "darwin") {
        exec(`open ${url}`);
    } else {
        exec(`xdg-open ${url}`);  // Assume Linux
    }
}

File: rick.dart
import 'dart:io';

void main() async {
  var result = await Process.run('curl', ['ascii.live/rick']);
  
  if (result.exitCode == 0) {
    print(result.stdout); // Print command output
  } else {
    print("Command failed: ${result.stderr}");
    openRickroll();
  }
}

void openRickroll() {
  var url = 'https://www.youtube.com/watch?v=dQw4w9WgXcQ';
  if (Platform.isWindows) {
    Process.run('start', [url], mode: ProcessStartMode.detached);
  } else if (Platform.isMacOS) {
    Process.run('open', [url], mode: ProcessStartMode.detached);
  } else {
    Process.run('xdg-open', [url], mode: ProcessStartMode.detached); // Assume Linux
  }
}